README.md

```diff
@@ -402,14 +402,69 @@ To learn by examples, see `examples/scripts` to have a feeling of how to script
+Which log messages are emitted is controlled by the `DEBUG` environment variable.
 
-The logs in the library follows those best practice rules (see [here](https://github.com/eclipse/thingweb.node-wot/issues/229)):
+In the following, we will show a couple of examples for its usage using wildcard characters (`*`).
+Note, however, that the syntax for setting an environment variable depends on your operating system and the terminal you use.
+See the `debug` documentation linked above for more details on platform-specific differences.
 
-1. Tag log messages with the package as following: `console.debug("[package-name]", "log message)`. This is useful to identify which package generated the log.
-2. Avoid to use `info` and `log` in packages other than the cli package.
+#### Examples
 
-Please follows these rules if you are going to contribute to node-wot library.
+First, you can enable all log messages by setting `DEBUG` to a wildcard like so:
+
+```sh
+DEBUG=* npm start
+```
+
+To only show `node-wot`-specific logging messages, prefix the wildcard with `node-wot`:
+
+```sh
+DEBUG=node-wot* npm start
+```
+
+To only show a specific log level, use one of `info`, `debug`, `warn`, or `error` as the suffix.
+Note in this context that you can provide multiple values for `DEBUG`.
+For example, if you want to show only `debug` and `info` messages, you can use the following:
+
+```sh
+DEBUG='*debug,*info' npm start
+```
+
+Finally, you can choose to only display log messages from a specific `node-wot` package.
+For example, if you only want to see log messages for the `core` package, use the following:
+
+```sh
+DEBUG=node-wot:core* npm start
+```
+
+Using the log levels above, you can also apply more fine-grained parameters for logging.
+For instance, if you only want to see `error` messages from the `binding-coap` package, use this:
+
+```sh
+DEBUG=node-wot:binding-coap*debug npm start
+```
+
+#### Adding logging functionality to a package
+
+If you are contributing a new binding, please use the `createLoggers` function from the `core` package for adding logging functionality to your package.
+The function accepts an arbitrary number of arguments that will be mapped to `debug` namespaces.
+In the example below, the `createLoggers` function will map its arguments `binding-foo` and `foo-server` to the namespace `node-wot:binding-foo:foo-server`.
+The resulting functions `debug`, `info`, `warn`, and `error` will append their log-level to this namespace when creating the actual log message.
+This enables filtering as described in the section before.
+
+```ts
+import { createLoggers } from "@node-wot/core";
+const { debug, info, warn, error } = createLoggers("binding-foo", "foo-server");
+
+function startFoo() {
+    info("This is an info message!");
+    debug("This is a debug message!");
+    warn("This is a warn message!");
+    error("This is an error message!");
+}
+```
```

Comment:
This content should be inside the Contributing.md file and here we should only give a referenced to the relevant section. We can move the section later.

Reply:
Oh, yeah, that is a good idea! :)

Reply:
Hmm, I suppose we could add a new section like "Adding a new Protocol Binding" to the `Contributing.md` for this purpose?

Reply:
> Hmm, I suppose we could add this to a new section like "Adding a new Protocol Binding", containing also more general instructions for implementors?

Yes, good point!

Reply:
Makes sense to move.